src/EventListener/AddFormatListener.php

```diff
@@ -50,11 +53,18 @@ public function onKernelRequest(GetResponseEvent $event)
         $mimeType = $requestFormat ? $request->getMimeType($requestFormat) : null;
         if (null === $mimeType || !isset($this->mimeTypes[$mimeType])) {
             if (null === $accept = $request->headers->get('Accept')) {
+                if (null !== $requestFormat) {
+                    throw $this->getNotAcceptableHttpException(null === $accept ? 'unknown' : $accept);
```

Comment:
Anyway, I don't think the logic here is correct:

1. If the MIME type is null or unknown
2. If the Accept request header is not set
3. If there's a request format specified\*

\* Correct me if I'm wrong, but looking at
https://github.com/symfony/http-foundation/blob/472367d63660b5f1426b33b87ba0558b3487945a/Request.php#L1377-L1397
and referring to
http://symfony.com/doc/current/quick_tour/the_controller.html#using-formats
this doesn't seem like what we intend to check - the format **requirement**.